library/core/src/slice/mod.rs

```diff
@@ -3059,6 +3059,30 @@ impl<T> [T] {
 
         left
     }
+
+    /// Moves `value` into `self[index]`, returning the old value.
+    ///
+    /// Neither value is dropped.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `index` is out of bounds.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_replace)]
+    ///
+    /// let mut v = [1, 2, 3];
+    /// let r = v.replace(1, 20);
+    ///
+    /// assert_eq!(v, [1, 20, 3]);
+    /// assert_eq!(r, 2);
+    /// ```
+    #[unstable(feature = "slice_replace", reason = "new API", issue = "none")]
+    pub fn replace(&mut self, index: usize, value: T) -> T {
```

Comment:
nit: don't bother making this change until we find out whether libs wants to move forward, but note that `mem::replace` is `must_use`, so this one probably should be too -- maybe something like
```rust
#[must_use = "if you don't need the old value, you can just assign the new value directly with `*slice[index] = value;`"]
```